velox/vector/tests/DecodedVectorTest.cpp

```diff
@@ -614,6 +614,34 @@ TEST_F(DecodedVectorTest, dictionaryOverLazy) {
   }
 }
 
+TEST_F(DecodedVectorTest, nestedLazy) {
```

Comment:
Can you also add a test in LazyVectorTest that verifies, LazyVector::ensureLoadedRows() and isLazyNotLoaded() works correctly for this type of row vector. (See lazyInDictionary for reference).